java-checks/src/main/java/org/sonar/java/checks/ConfusingOverloadCheck.java

```diff
@@ -0,0 +1,130 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Sets;
+import org.apache.commons.lang.BooleanUtils;
+import org.sonar.api.server.rule.RulesDefinition;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.java.model.declaration.MethodTreeImpl;
+import org.sonar.java.resolve.JavaSymbol.MethodJavaSymbol;
+import org.sonar.java.resolve.JavaType;
+import org.sonar.java.tag.Tag;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.semantic.Type;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.squidbridge.annotations.ActivatedByDefault;
+import org.sonar.squidbridge.annotations.SqaleConstantRemediation;
+import org.sonar.squidbridge.annotations.SqaleSubCharacteristic;
+
+import java.util.List;
+import java.util.Set;
+
+import static org.sonar.plugins.java.api.semantic.Symbol.MethodSymbol;
+import static org.sonar.plugins.java.api.semantic.Symbol.TypeSymbol;
+
+@Rule(
+  key = "S2177",
+  name = "Child class methods named for parent class methods should be overrides",
+  priority = Priority.MAJOR,
+  tags = {Tag.PITFALL})
+@ActivatedByDefault
+@SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.ARCHITECTURE_RELIABILITY)
+@SqaleConstantRemediation("30min")
+public class ConfusingOverloadCheck extends IssuableSubscriptionVisitor {
+  private static final Set<String> SERIALIZATION_METHOD_NAME = Sets.newHashSet("writeObject", "readObject", "readObjectNoData", "writeReplace", "readResolve");
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return ImmutableList.of(Tree.Kind.METHOD);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    if (!hasSemantic()) {
+      return;
+    }
+    MethodTreeImpl methodTree = (MethodTreeImpl) tree;
+    if (BooleanUtils.isFalse(methodTree.isOverriding())) {
+      MethodSymbol methodSymbol = methodTree.symbol();
+      TypeSymbol owner = (TypeSymbol) methodSymbol.owner();
+      Type superClass = owner.superClass();
+      if(superClass != null && !SERIALIZATION_METHOD_NAME.contains(methodSymbol.name())) {
+        checkMethod(methodTree.simpleName(), methodSymbol, superClass);
+        superClass = superClass.symbol().superClass();
+        while (superClass != null) {
+          checkStaticMethod(methodTree.simpleName(), methodSymbol, superClass);
```

Comment:
If there is several methods in the hierarchy, this call will raise too many issue.